portfolio/src/main/java/com/google/sps/servlets/FollowedTagsServlet.java

```diff
@@ -0,0 +1,93 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.blobstore.BlobInfo;
+import com.google.appengine.api.blobstore.BlobInfoFactory;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.images.ImagesService;
+import com.google.appengine.api.images.ImagesServiceFactory;
+import com.google.appengine.api.images.ServingUrlOptions;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
+import com.google.sps.data.BlogMessage;
+import com.google.sps.data.BlogHashMap;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.*;
+import java.net.MalformedURLException;
+import java.net.URL;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+@WebServlet("/follow-tags")
+public class FollowedTagsServlet extends HttpServlet {
+    List<String> followedPosts= new ArrayList<>();
+    
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    }
+    
+
+    /**
+   * Converts a ServerStats instance into a JSON string using the Gson library
+   */
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      UserService userService = UserServiceFactory.getUserService();
+     
+      String commentType = getParameter(request, "tags", "#general");
```

Comment:
i think we don't need the getParameter function since a user can decide to follow tags or not. If they choose not to we don't want to assign a default tag, because then we'll record that they follow the default tag even when they actually dont.

Reply:
If they don't follow any tag we can just return so that nothing is saved in datastore

Reply:
QQ about this - how are you going to differentiate between a user who is following no tags and user who hasn't selected the tags they want to follow yet? Or a user who's request to follow tags go lost?

Reply:
I'm not sure how to handle a lost request but as far as the back end is concerned, right now there will either be a one or more instance of a user who has a tag that they follow in the datastore or there will be no instance of that user. But a user can exist outside of the datastore.

Reply:
For lost request - this is just if someone responds with the tags they want to follow but the request gets lost on the way to the backend. Essentially like that request never happened.

This isn't something you need to solve in this PR, but I think the plan was to have a popup/redirect to ask users to put in their tag preference. If a user doesn't put in their tag preference, do you want the popup to show every time they visit the site? If not, you will need a way to track the fact that a user doesn't want to follow any tags.